web/chromecom.js

```diff
@@ -293,46 +293,48 @@ function setReferer(url, callback) {
 // chrome.storage.local to chrome.storage.sync when needed.
 var storageArea = chrome.storage.sync || chrome.storage.local;
 
-Preferences._writeToStorage = function (prefObj) {
-  return new Promise(function (resolve) {
-    if (prefObj === Preferences.defaults) {
-      var keysToRemove = Object.keys(Preferences.defaults);
-      // If the storage is reset, remove the keys so that the values from
-      // managed storage are applied again.
-      storageArea.remove(keysToRemove, function() {
-        resolve();
-      });
-    } else {
-      storageArea.set(prefObj, function() {
-        resolve();
-      });
-    }
-  });
-};
-
-Preferences._readFromStorage = function (prefObj) {
-  return new Promise(function (resolve) {
-    if (chrome.storage.managed) {
-      // Get preferences as set by the system administrator.
-      // See extensions/chromium/preferences_schema.json for more information.
-      // These preferences can be overridden by the user.
-      chrome.storage.managed.get(Preferences.defaults, getPreferences);
-    } else {
-      // Managed storage not supported, e.g. in old Chromium versions.
-      getPreferences(Preferences.defaults);
-    }
+class ChromePreferences extends BasePreferences {
+  _writeToStorage(prefObj) {
+    return new Promise((resolve) => {
+      if (prefObj === this.defaults) {
+        var keysToRemove = Object.keys(this.defaults);
+        // If the storage is reset, remove the keys so that the values from
+        // managed storage are applied again.
+        storageArea.remove(keysToRemove, function() {
+          resolve();
+        });
+      } else {
+        storageArea.set(prefObj, function() {
+          resolve();
+        });
+      }
+    });
+  }
 
-    function getPreferences(defaultPrefs) {
-      if (chrome.runtime.lastError) {
-        // Managed storage not supported, e.g. in Opera.
-        defaultPrefs = Preferences.defaults;
+  _readFromStorage(prefObj) {
+    return new Promise((resolve) => {
+      if (chrome.storage.managed) {
+        // Get preferences as set by the system administrator.
+        // See extensions/chromium/preferences_schema.json for more information.
+        // These preferences can be overridden by the user.
+        chrome.storage.managed.get(this.defaults, getPreferences);
+      } else {
+        // Managed storage not supported, e.g. in old Chromium versions.
+        getPreferences(this.defaults);
       }
-      storageArea.get(defaultPrefs, function(readPrefs) {
-        resolve(readPrefs);
-      });
-    }
-  });
-};
+
+      var getPreferences = (defaultPrefs) => {
```

Comment:
Isn't it a problem that this is defined *after* it is used? I think it's better to move this above the `if`.